doc/optional-state.rst

```diff
@@ -0,0 +1,74 @@
+Optional per-peer state
+=======================
+
+An Autocrypt-capable client must store some specific state about each
+of its peers.
+
+This document attempts to describe additional optional state that can
+improve the user experience in some corner cases.
+
+Unlike the standard Autocrypt level 0 state management, some of these
+rules depend on a client being able to keep track of whether it has
+seen a given message before or not, and these guidelines may cause
+non-deterministic results depending on the order that messages are
+encountered.
+
+Additional state
+----------------
+
+Optionally, an agent MAY store and maintain the following data in
+order to provide clearer feedback to the user:
+
+* ``counting_since``: The UTC timestamp of when we started counting
+* ``count_have_ach``: A count of parsed AutoCrypt headers
+* ``count_no_ach``: A count of messages without AutoCrypt headers
+* ``bad_user_agent``: The apparent user-agent (if known) of the last
+  message seen without AutoCrypt headers.
+
+.. note::
+
+     These attributes are all optional, and are presented here as a
```

Comment:
I feel like this paragraph has a lot of redundancy in it, and its actual content (what this is for) should come before listing attributes. (also see comment above)